### DISCUSSION TITLE
Onboarding to the API assorted problems

### DISCUSSION THREAD

**Discussion by AdsAPIPM** [amzn/ads-advanced-tools-docs#161] [General]
I keep getting this error when I try to complete step "Create Authorization Grant" in the [onboarding guide](https://advertising.amazon.com/API/docs/en-us/guides/get-started/create-authorization-grant). Anyone has any ideas on what I am doing wrong?

**Comment by jerirose** [comment]
Hi @AdsAPIPM , the "unknown scope" error is most likely to arise in three cases:
- The LwA client ID is not approved to use the requested scope -- if you have been approved to use the Ads API, make sure that you have [assigned API access to your LwA app](https://advertising.amazon.com/API/docs/en-us/guides/onboarding/assign-api-access)
- The string provided for the `scope` parameter does not precisely match the scope, e.g, `advertising:campaign_management` (with one colon) instead of `advertising::campaign_management`
- No `scope` parameter was included in the request

Do any of these match your case?

**Comment by AdsAPIPM** [comment]
As it turns out. I skipped a step. This one: https://advertising.amazon.com/API/docs/en-us/guides/onboarding/assign-api-access

**Comment by AdsAPIPM** [comment]
also word of caution. In the step above "Assign API access to a Login with Amazon application". Users need to be logged in with the same email they used to apply for API access. 

As described here: https://advertising.amazon.com/API/docs/en-us/guides/onboarding/apply-for-access

**Comment by AdsAPIPM** [comment]
I have a new problem... Now I keep getting

"{"code":"UNAUTHORIZED","details":"HTTP 401 Unauthorized","requestId":""}"

When I try to get a profile. I can refresh tokens no problem. Anyone else have had that issue before? What am I doing wrong?

**Comment by jerirose** [comment]
@AdsAPIPM The only required authorization headers for the profiles endpoint are:

- **Amazon-Advertising-API-ClientId**: Your Login With Amazon client ID
- **Authorization**: The string "bearer " (including the space) prepended to a valid access token associated with the client ID

For access token "Atza|zZzZzZzZ", the value for the **Authorization** header would be "bearer Atza|zZzZzZzZ".

Ensure that these headers are included in your request and spelled precisely as above, and that the access token is current (access tokens expire after 60 minutes). If these headers are correctly provided, even a request using an account with no profiles in the region should return an empty array `[]` rather than an unauthorized response.

**Reply by SARANGGULVE** [reply]
I also got the same problem @AdsAPIPM this friends can you plz help me

**Comment by AdsAPIPM** [comment]
This solved my problem. Also the Postman collections are gold

**Comment by davidchapuis** [comment]
Hi folks, running into the same issue here
@AdsAPIPM could you solve it with the account you also used to request API access or another developer account?
@jerirose Is it possible to go through the authorization grant workflow with any developer account in the team or it only can be done with the account tied to the email that requested API access in the first place?

**Comment by jerirose** [comment]
Hi @davidchapuis , the **Assign API access** step should be completed with the same Amazon user account used to request API access. If the link was followed by a different logged-in user, please reach out to the API support team via Zendesk as described on the [Support page](https://advertising.amazon.com/API/docs/en-us/support/overview).

One troubleshooting step that was not available at the time of this initial conversation is the [Developer dashboard](https://advertising.amazon.com/developer/overview) in the advanced tools center. If you visit that URL while logged into your Amazon Developer account, you should see a list of LwA apps associated with that account, along with a list of available permission scopes for each. If you see the expected client ID listed, and it includes the `advertising::campaign_management` scope, then any "Unknown scope" error is likely related to one of the other issues listed above.

**Reply by davidchapuis** [reply]
Hi @jerirose thanks so much for sharing this link, very useful!
My current scopes are: profile, profile:contact, profile:mobile_number, profile:name
There are no ads_api scopes although the access request has been done some time ago already.
How much time it can take to have ads_api scopes added?
Shall I just send a new ads API access request?

**Reply by drishyajose** [reply]
Hi @davidchapuis ,
I’m also facing the same issue with no scopes available other than profile:contact, profile, profile:name, and profile:mobile_number.

How did you resolve it?

I had clicked on the Amazon onboarding link, but it always shows an error. I’ve been contacting Amazon Ads Support continuously for a week, but it’s still not resolved.

Any help would be greatly appreciated. 
Thank you!
![api](https://github.com/user-attachments/assets/42a88f2e-47a4-4ed7-84f2-ff9930c395e6)

**Reply by devchef87** [reply]
Did you ever get this resolved, im running into the exact same issue. I click the link and says I don't have access to the ads API scope , even though I was approved. And no way to assign access. Ive been reaching out to support and have yet to hear back. Maybe @jerirose  can offer some guidance?

**Reply by jerirose** [reply]
Hi @devchef87 , most frequently when this occurs, it's a misstep on the **Assign API access** step in the onboarding flow -- customers often complete that step while logged into a different Amazon user account than intended. If you're seeing similar output in the **Your apps** tab to what @drishyajose posted above, unfortunately the only way to reassign the scope to the correct client is through the support team.

Did you reach out to support via one of the addresses listed on the [Support page](https://advertising.amazon.com/API/docs/en-us/support/overview)? `ads-api-onboarding@amazon.com` should be the correct address to use for this issue.
